gdc_ng_models/models/entity_set.py

```diff
@@ -0,0 +1,108 @@
+  by the cohort service in the backend system.
+"""
+
+import sqlalchemy
+from sqlalchemy.dialects import postgresql
+from sqlalchemy.ext import declarative
+from gdc_ng_models.models import accessed, audit
+
+Base = declarative.declarative_base()
+
+
+class EntitySet(Base, audit.AuditColumnsMixin, accessed.AccessedColumnMixin):
+    """A base definition for the entity_set.
+
+    Attributes:
+       id: A unique string identifier for the entity_set defined by the client
+       type: Ephmeral, Frozen, Mutable
+       entity_type: case, file, gene, ssm used to route to the indices in elasticsearch
+         e.g., gdc_case_set, gdc_file_set, gdc_gene_set, gdc_ssm_set
+       entity_ids: array of node UUIDs of entity_type
+       created_datetime: The date and time when the record is created.
+       updated_datetime: The date and time when the record is last updated.
+       accessed_datetime: The date and time when the record is last accessed.
+    """
+
+    __tablename__ = "entity_set"
+
+    # Requirement: custom IDs
+    # Requirement: support sha256 outputlength (64hex characters) and longest id as identified by the front end.
+    # No default ID, all clients will provide their own IDs for consistent usage
+    id = sqlalchemy.Column(
+        sqlalchemy.String(128),
+        primary_key=True,
+    )
+    type = sqlalchemy.Column(
+        postgresql.ENUM("ephmeral", "frozen", "mutable", name="entity_set_type"),
+        nullable=False,
+    )
+    entity_type = sqlalchemy.Column(
+        postgresql.ENUM("case", "file", "gene", "ssm", name="entity_type"),
+        nullable=False,
+    )
+
+    # entity_ids are UUIDs that are 36 characters long.
+    # Postgres is more performant when the data sizes are not dynamic
+    entity_ids = sqlalchemy.Column(
+        postgresql.ARRAY(sqlalchemy.String(36)), nullable=False
+    )
+
+    def __repr__(self):
+        return (
+            "<EntitySet("
+            "id={id}, "
+            "type='{type}', "
+            "entity_type={entity_type}, "
+            "entity_ids={entity_ids}"
+            "created_datetime={created_datetime}, "
+            "updated_datetime={updated_datetime}), "
+            "accessed_datetime={accessed_datetime})>".format(
+                id=self.id,
+                type=self.entity_type,
+                entity_type=self.entity_type,
+                entity_ids=self.entity_ids,
+                created_datetime=self.created_datetime.isoformat()
+                if self.created_datetime
+                else None,
+                updated_datetime=self.updated_datetime.isoformat()
+                if self.updated_datetime
+                else None,
+                accessed_datetime=self.accessed_datetime.isoformat()
+                if self.accessed_datetime
+                else None,
+            )
+        )
+
+    def to_json(self):
+        return {
+            "id": str(self.id),
+            "type": self.type,
```

Comment:
Is casting necessary here?
```suggestion
            "type": str(self.type),
```

Reply:
It passed the unit test alright... but I'll make that change so it matches the str(self.id)

Reply:
It may not be necessary, I just figured if you did it for one enum type it would make sense to keep it consistent. Removing the casting from entity_type, if not necessary, is also an option.